tools/freeze-deps.py

```diff
@@ -73,18 +81,33 @@
             defs.append(line)
     output.writestr(zinfo, "\n".join(defs))
 
-# Copy the compat.bzl if it was added to the maven install via generate_compat_repositories = True attribute
-# need to update the repo name in the file.
-compat_bzl = path.join(base, "external", args.repo, "compat.bzl")
-if path.exists(compat_bzl):
-    zinfo = zipfile.ZipInfo(filename="compat.bzl", date_time=(1980, 1, 1, 0, 0, 0))
-    libname = args.zip_repo if args.zip_repo else path.basename(path.splitext(args.zip)[0])
-    with open(compat_bzl) as f:
-        lines = [re.sub(args.repo, libname, line) for line in f.read().split('\n')]
-    output.writestr(zinfo, "\n".join(lines))
-
+# Generate a compat.bzl file. This is only needed for workspace-based builds, but the
+# file is not generated in a `bzlmod`-created `@maven`.
+lock_file = root / "imported_maven_install.json"
+
+compat_content = """load("@contrib_rules_jvm_deps//:compat_repository.bzl", "compat_repository")
+
+def compat_repositories():
+    """
```

Comment:
do you want to add a `pass` here for the super-weird case of there not being any artifacts?

Reply:
I don't think we'd be freezing deps if we have no artifacts….